src/ctelnet.cpp

```diff
@@ -1804,18 +1804,83 @@ void cTelnet::processTelnetCommand(const std::string& telnetCommand)
+                            /*
+                            * The valid characters for termTerm are more restricted
+                            * than being ASCII - from:
+                            * https://tools.ietf.org/html/rfc1010 (page 29):
+                            * "A terminal names may be up to 40 characters taken from
+                            * the set of uppercase letters, digits, and the two
+                            * punctuation characters hyphen and slash.  It must start
+                            * with a letter, and end with a letter or digit."
+                            * Once we comply with that we can be certain that Mud
+                            * Server encoding will NOT be an issue!
+                            */
+                            static const auto allInvalidCharacters = QRegularExpression(qsl("[^A-Z,0-9,-,\\/]"));
+                            static const auto multipleHyphens = QRegularExpression(qsl("-{2,}"));
+                            QString sanitisedTermType = termType.toUpper()
+                                                                .replace(QChar('.'), QChar('/'))
+                                                                .replace(QChar::Space, QChar('-'))
+                                                                .replace(allInvalidCharacters, QChar('-'))
+                                                                .replace(multipleHyphens, QChar('-'))
+                                                                .left(40)
+                                                            .toLatin1().constData();
+
+                            for (int i = sanitisedTermType.size() - 1; i >= 0; --i) {
+                                if (sanitisedTermType[i].isLetterOrNumber()) {
+                                    sanitisedTermType = sanitisedTermType.left(i + 1);
+                                    break;
+                                }
+                            }
+
+                            Q_ASSERT_X(!sanitisedTermType.isEmpty(),
+                                       "cTelnet::processTelnetCommand(...)",
+                                       "ended up with an empty version string whilst trying to sanitise the Mudlet one");
+
+                            cmd += sanitisedTermType.toLatin1().constData();
+
+                            if (!mpHost->mFORCE_MTTS_NEGOTIATION_OFF) { // If we don't MTTS, remainder of the cases do not execute.
+                                mCycleCountMTTS++;
+                            }
+
+                            break;
+                        }
+                        case 1:
+                            qDebug() << "MTTS enabled";
+                            cmd += qsl("XTERM-256COLOR").toUtf8().constData(); // DUMB, ANSI, VT100, XTERM
+                            mCycleCountMTTS++;
+                            qDebug() << "WE send MTTS terminal type is XTERM-256COLOR";
```

Comment:
Mudlet does not support the cursor movement commands and some of the other features that Scandum lists at: https://tintin.mudhalla.net/info/vt100/  so we should not use `XTERM` or `VT100` - we should instead claim `ANSI` compatibility - also we do support 16M colour modes so instead this (ASCII character set - so can be encoded quicker with the Latin1 rather than the UTF-8 encoding) line should be:
```suggestion
                            cmd += qsl("ANSI-TRUECOLOR").toLatin1().constData(); // DUMB, ANSI, VT100, XTERM
                            mCycleCountMTTS++;
                            qDebug() << "WE send MTTS terminal type is ANSI-TRUECOLOR";
```